creator/ingest_runs/mutations.py

```diff
@@ -56,19 +55,39 @@ def mutate(self, info, input):
         with transaction.atomic():
             ingest_run = IngestRun()
             ingest_run.save()
+            ingest_run.creator = user
             ingest_run.versions.set(versions)
             ingest_run.save()
 
-        uuid_str = str(ingest_run.id)
-        django_rq.enqueue(
-            run_ingest,
-            args=(uuid_str,),
-            job_id=uuid_str,
-        )
-
+        # When IngestRuns are created, they're in the waiting state, so we
```

Comment:
Not sure where to put this so just putting it here. Could you do this instead:
```python
        # Create ingest run for a set of file versions, if one does not exist
        # with same input_hash
        ingest_run = IngestRun()
        input_hash = ingest_run.compute_input_hash(versions=versions)
        
        # Check if duplicate ingest runs exist 
        irs_with_same_input_hash = IngestRun.objects.filter(
            state__in=["waiting", "started"],
            input_hash=ingest_run.input_hash,
        ).all()

        if irs_with_same_input_hash.count() > 0:
            cancel_duplicate_ingest_runs(irs_with_same_input_hash)
        else:
            with transaction.atomic():
                ingest_run.save()
                ingest_run.creator = user
                ingest_run.versions.set(versions)
                ingest_run.save()

            ingest_run.queue.enqueue(
                run_ingest,
                args=(ingest_run.id,),
                job_id=str(ingest_run.id),
            )
        return StartIngestRunMutation(ingest_run=ingest_run)
```

This way you don't have to save the run to the db and then delete it if you find duplicates. Also, that ^ requires a slight mod to the `IngestRun.compute_hash` method:

```python
    def compute_input_hash(self, versions=None):
        """
        Compute an MD5 hash digest from the IngestRun's input parameters:
            - List of file version ids (Version.kf_id)
        """
        versions = versions or self.versions.all()
        version_id_str = "".join(sorted(v.kf_id for v in versions))
        return hashlib.md5(version_id_str.encode("utf-8")).hexdigest()
```

Reply:
ah yes this makes sense. i was trying to figure out how to do it without saving and deleting. but didn't think to change the `compute_hash` method

Reply:
Need to do 
```
# Check if duplicate ingest runs exist 
        irs_with_same_input_hash = IngestRun.objects.filter(
            state__in=["waiting", "started"],
            input_hash=input_hash,
        ).all()
```
instead and it works